modules/liblc3/CMakeLists.txt

```diff
@@ -1,22 +1,32 @@
 if(CONFIG_LIBLC3)
-
 zephyr_library_named(liblc3)
-zephyr_library_compile_options(-O3 -std=c11 -ffast-math -Wno-array-bounds)
+zephyr_library_compile_options(-O3 -std=c11 -Wall -ffast-math -Wextra -Wdouble-promotion -Wvla -pedantic)
```

Comment:
Why the chance in compile options? Requirement?

Reply:
> the CI compiler failed due to access outside array bounds (-Warray-bounds)

That doesn't sound good :D Do you remember what the issue was? Any chance that it's an actual issue in the liblc3?

Reply:
It appears to be false positive. It seems that the warning was explicitly suppressed for clang https://github.com/google/liblc3/blob/bb85f7dde4195bfc0fca9e9c7c2eed0f8694203c/src/common.h#L66.
If this does not show up for me but shows in CI I suspect it's because of difference in gcc version https://gcc.gnu.org/bugzilla/show_bug.cgi?id=100038